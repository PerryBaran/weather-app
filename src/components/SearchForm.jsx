import React, { useRef } from "react";
import PropTypes from "prop-types";

function SearchForm({ handleSearch }) {
  const searchRef = useRef(null);

  const handleSubmit = () => {
    handleSearch(searchRef.current.value);
  };

  return (
    <div className="search-form">
      <input type="text" ref={searchRef} />
      <button type="submit" onClick={handleSubmit}>
        Search
      </button>
    </div>
  );
}

SearchForm.propTypes = {
  handleSearch: PropTypes.func.isRequired,
};

export default SearchForm;
