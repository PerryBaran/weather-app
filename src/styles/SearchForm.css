.search-form {
  height: 25px;
  width: calc(100% - 50px);
  max-width: 600px;
  margin: auto 0;
  background-color: white;
}

.search-form input {
  all: unset;
  box-sizing: border-box;
  height: 100%;
  width: calc(100% - 65px);
  padding: 5px;
  font-size: 1.2rem;
}

.search-form button {
  all: unset;
  height: 100%;
  width: 65px;
  font-size: 1rem;
  text-align: center;
  font-weight: 400;
  vertical-align: top;  
  background-color: var(--dark-blue);
  color: var(--almost-white);
  text-shadow: none;
  cursor: pointer;
}