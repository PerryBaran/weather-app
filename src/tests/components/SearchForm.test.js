import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    handleSearch: () => {},
    setSelectedUnits: () => {},
  };

  describe("snapshot", () => {
    it("renders correctly", () => {
      const { asFragment } = render(
        <SearchForm
          handleSearch={validProps.handleSearch}
          setSelectedUnits={validProps.setSelectedUnits}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("button", () => {
    it("renders corret text", () => {
      const { getByText, getByLabelText } = render(
        <SearchForm
          handleSearch={validProps.handleSearch}
          setSelectedUnits={validProps.setSelectedUnits}
        />
      );

      const metricRadio = getByLabelText("Metric:");
      const imperialRadio = getByLabelText("Imperial:");

      expect(getByText("Search")).toHaveAttribute("type", "submit");
      expect(metricRadio).toHaveAttribute("type", "radio");
      expect(metricRadio).toHaveAttribute("id", "metric");
      expect(imperialRadio).toHaveAttribute("type", "radio");
      expect(imperialRadio).toHaveAttribute("id", "imperial");
    });
  });
});
