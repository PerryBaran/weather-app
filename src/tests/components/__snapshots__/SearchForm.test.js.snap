// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm snapshot renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <div
      class="search-form__search"
    >
      <input
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="search-form__radio"
    >
      <h3>
        Units:
      </h3>
      <label
        class="search-form__radio__label"
        for="metric"
      >
        Metric
        <input
          id="metric"
          type="radio"
        />
      </label>
      <label
        class="search-form__radio__label"
        for="imperial"
      >
        Imperial
        <input
          id="imperial"
          type="radio"
        />
      </label>
    </div>
  </form>
</DocumentFragment>
`;
